proposals/lip-0051.md

```diff
@@ -769,14 +770,227 @@ if the above send function does not fail:
+
+* Initial checks on the properties of `genesisBlockAssetObject`:
+   * Accross all elements of the `userSubstore` array, all `address` values must have length `ADDRESS_LENGTH`. 
+   * Accross all elements of the `userSubstore` array, the pairs `address, tokenID` must be unique (each pair appears at most once, though a given address can appear multiple times with different token IDs). 
+   * For each such `address, tokenID` pair, the values used as `moduleID` in the `lockedBalances` array must be unique.
+   * Accross all element of the `supplySubstore` array, all values given for `localID` must be unique.
+   * Accross all elements of the `escrowSubstore` array, the pairs `escrowChainID, localID` must be unique (each pair appears at most once). 
+   * In the `terminatedEscrowSubstore` array, each element must be unique (appears at most once). 
+
+* For each entry `userEntry` in `genesisBlockAssetObject.userSubstore`, create an entry in the user substore  with
+  ```python
+  storeKey = userEntry.address || uint32be(userEntry.tokenID.chainID) || uint32be(userEntry.tokenID.localID)
+  storeValue = {"availableBalance": userEntry.availableBalance,
+                "lockedBalances": [{"moduleID": lockedBalance.moduleID,
+                                    "amount": lockedBalance.amount
+                                   } for each lockedBalance in userEntry.lockedBalances]
+               } serialized using userStoreSchema.
+  ```
+
+* For each entry `supplyEntry` in `genesisBlockAssetObject.supplySubstore`, create an entry in the supply substore  with
+  ```python
+  storeKey = uint32be(supplyEntry.localID)
+  storeValue = {"totalSupply": supplyEntry.totalSupply} serialized using supplyStoreSchema.
+  ```
+
+* For each entry `escrowEntry` in `genesisBlockAssetObject.escrowSubstore`, create an entry in the escrow substore  with
+  ```python
+  storeKey = uint32be(escrowEntry.escrowChainID) || uint32be(escrowEntry.localID)
+  storeValue = {"amount": escrowEntry.amount} serialized using escrowStoreSchema.
+  ```
+
+* Create an entry in the available local ID substore with
+  ```python
+  storeKey = EMPTY_BYTES
+  storeValue = {"nextAvailableLocalID": genesisBlockAssetObject.availableLocalIDSubstore} 
+               serialized using availableLocalIDStoreSchema.
+  ```
+
+* For each entry `terminatedChainID` in `genesisBlockAssetObject.terminatedEscrowSubstore`, create an entry in the terminated escrow substore with
+  ```python
+  storeKey = uint32be(terminatedChainID)
+  storeValue = {"escrowTerminated": True} serialized using terminatedEscrowSchema.
+  ```
+  
+Once the module store is initialized, its validity is attested via the two checks below.
+* Check that for each native token the total supply is correct. 
+  This can be done by checking that the function below returns `True`.
+  ```python
+  totalSupplyCorrect():
```

Comment:
Maybe `areTotalSuppliesCorrect()` or `validateSupplyStoreEntries()`. The current name would be an odd phrase in English.